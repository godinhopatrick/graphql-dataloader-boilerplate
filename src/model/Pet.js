// @flow

import mongoose from 'mongoose';

const Schema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    age: {
      type: Number,
      required: true,
    },
    animal: {
      type: String,
      default: 'DOG',
    },
    breed: {
      type: String,
    },
  },
  {
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    },
    collection: 'pet',
  },
);

export default mongoose.model('Pet', Schema);
