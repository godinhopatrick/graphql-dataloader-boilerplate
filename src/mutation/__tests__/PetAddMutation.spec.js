import { graphql } from 'graphql';
import { schema } from '../../schema';
import { setupTest, getContext } from '../../../test/helper';

import { User } from '../../model';
import { Pet } from '../../model';

beforeEach(async () => await setupTest());

it('should not allow anonymous user', async () => {
  const query = `
    mutation M {
      PetAdd(input: {
        name: "Teste value"
        age: 20
        animal: "Cat"
        breed: "Nojento"
      }) {
        petEdge {
          node {
            name
            age
            animal
            breed
          }
        }
      }
    }
  `;

  const rootValue = {};
  // No user should be passed to context since we are testing an anonymous session
  const context = getContext();

  const result = await graphql(schema, query, rootValue, context);

  expect(result).toMatchSnapshot();
});

it('should create a record on database', async () => {
  const user = new User({
    name: 'user',
    email: 'user@example.com',
  });

  await user.save();

  const query = `
    mutation M {
      PetAdd(input: {
        name: "Example value"
        age: "Example value"
        animal: "Example value"
        breed: "Example value"
      }) {
        petEdge {
          node {
            name
            age
            animal
            breed
          }
        }
      }
    }
  `;

  const rootValue = {};
  const context = getContext({ user });

  const result = await graphql(schema, query, rootValue, context);

  expect(result).toMatchSnapshot();
});
