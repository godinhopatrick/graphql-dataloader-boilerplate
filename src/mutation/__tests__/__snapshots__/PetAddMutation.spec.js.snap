// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a record on database 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Cannot query field "PetAdd" on type "Mutation".],
  ],
}
`;

exports[`should not allow anonymous user 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Cannot query field "PetAdd" on type "Mutation".],
  ],
}
`;
